projects/packages/jetpack-mu-wpcom/tests/php/features/launchpad/class-launchpad-task-list-validation-test.php

```diff
@@ -91,11 +107,17 @@ public function provide_validate_task_list_test_cases() {
 	 *
 	 * @param array $task_list       Task list to validate.
 	 * @param bool  $expected_result Expected validation result.
+	 * @param bool  $expected_is_wp_error Flag to signal if the expected result is a WP_Error.
 	 * @dataProvider provide_validate_task_list_test_cases
 	 */
-	public function test_validate_task_list( $task_list, $expected_result ) {
+	public function test_validate_task_list( $task_list, $expected_result, $expected_is_wp_error ) {
 		$result = Launchpad_Task_Lists::validate_task_list( $task_list );
 
+		if ( $expected_is_wp_error ) {
+			$this->assertTrue( is_wp_error( $result ) );
+			return;
+		}
+
 		$this->assertSame( $expected_result, $result );
```

Comment:
Should we not do this in the opposite order so we always check `$expected_result`?
```suggestion
		$this->assertSame( $expected_result, $result );

		if ( $expected_is_wp_error ) {
			$this->assertTrue( is_wp_error( $result ) );
		}
```

Reply:
The issue with that is `assertSame` will fail because `$expected_result` and `$result` are technically different `WP_Error` objects.

Reply:
_But_, I can probably switch from using `assertSame` to checking the actual error message.

Reply:
Maybe we can write this as follows, where we can actually drop the `$expected_is_wp_error` argument:
```suggestion
		if ( is_wp_error( $expected_result ) ) {
			$this->assertTrue( is_wp_error( $result ) );
			$this->assertSame( $expected_result->get_error_code(), $result->get_error_code() );
			$this->assertSame( $expected_result->get_error_message(), $result->get_error_message() );
		} else {
			$this->assertSame( $expected_result, $result );
		}
```